src/kmi/simple.c

```diff
@@ -193,20 +193,20 @@ int simple_parse_opts(void *opt_args)
 			+ strlen((char *) key[i].dk) + strlen(":") + strlen((char *) key[i].dki) + 1;
 
 		if (priv.dk_list)
-			priv.dk_list = realloc(priv.dk_list, dk_list_len);
+			priv.dk_list = (char*)realloc(priv.dk_list, dk_list_len);
 		else
-			priv.dk_list = calloc(dk_list_len, sizeof(unsigned char));
+			priv.dk_list = (char*)calloc(dk_list_len, sizeof(unsigned char));
 		if (priv.dk_list == NULL) {
 			ltfsmsg(LTFS_ERR, 10001E, __FUNCTION__);
 			return -LTFS_NO_MEMORY;
 		}
 		*(priv.dk_list + original_dk_list_len) = '\0';
 
 		if (original_dk_list_len)
-			strcat((char *) priv.dk_list, "/");
-		strcat((char *) priv.dk_list, (char *) key[i].dk);
-		strcat((char *) priv.dk_list, ":");
-		strcat((char *) priv.dk_list, (char *) key[i].dki);
+			arch_strcat((char *) priv.dk_list, dk_list_len, "/");
```

Comment:
I cannot　understand why `strcat_s()` is needed here. Every input string can be handled safely in this function...

If you have a reason please explain it by comment and add `strcat_s()` function into  linux side. I don't like Annex K, but it is better than this, if you have a strong reason.